### PR TITLE
Proposal: Metadata Processing, Generic Apex Doc Annotation & indexOnly Parameter

### PR DESCRIPTION
**Pull request by HerrX2000**
- Changed behaviour of selecting the directories, now expects the command to be run in the root sfdx folder for better UX 
  - This is incompatible with the recent 1.15 change to accept an array of directories.
  - Feel free to take a look if you like the idea
  - Potentially breaking backwards compatibility depending on the defaults
- Added gitattributes to have better default lineendings with different OSes
- Added Metadata Processing
- Added Generic Apex Doc Annotation support
- Added indexOnly parameter to reduce execution time with pipelines
- Added Github Action

### CODE REVIEW

**Comment by HerrX2000** [comment]
Hi @cesarParra , feel free to hit me up, if you would like to include any of the mentioned features upstream.

File: src/cli/generate.ts [hunk 1]
```diff
@@ -3,18 +3,28 @@ import * as yargs from 'yargs';
...
 
 const argv = yargs.options({
   sourceDir: {
-    type: 'array',
+    type: 'string', //Upstream 1.4 supports arrays as input. Our fork will not because, we parse the directories differently.
     alias: 's',
-    demandOption: true,
-    describe: 'The directory or directories location(s) which contains your apex .cls classes.',
+    default: pwd,
+    describe: 'The root directory of your salesforce project. The sfdx-project.json determines the class folder.',
```
Comment (cesarParra) [inline src/cli/generate.ts:22]: I have a few thoughts around the SFDX change:

- Believe or not I still work with some legacy projects that haven't been migrated to SFDX (so they don't have an sfdx-project.json). Unfortunately these can't be migrated just yet (they can but it is easier to not touch them for different reasons: legacy code, 1st gen managed packages, etc.). I imagine that's the case for others out there so I'd still want this tool to be able to support those kind of projects
- The way the current tool works it traverses the directory structure recursively anyways, so even if they are SFDX projects they should still be supported. What would be the advantage of introducing this change?

File: src/cli/generate.ts [hunk 2]
```diff
@@ -50,15 +60,60 @@ const argv = yargs.options({
       'Define whether the generated files should be grouped by the @group tag on the top level classes.' +
       'If set to true, a sub directory will be created per group inside of the specified target directory.',
   },
+  indexOnly: {
+    type: 'boolean',
+    default: false,
+    describe:
+      'Defines whether only the index file should be  generated.',
```
Comment (cesarParra) [inline src/cli/generate.ts:67]: Minor nitpicky comment: There are 2 spaces in between "be" and "generated"

**Review by cesarParra — CHANGES_REQUESTED** [review]

File: src/cli/generate.ts (hunk 2 above)
Comment (HerrX2000) [reply src/cli/generate.ts]: Fixed

**Comment by HerrX2000** [comment]
Our local version is not yet update to the upstream 1.16 version so I resolved the merge conflicts here by hand.

**Review by cesarParra — APPROVED** [review]
Looks great

**Comment by cesarParra** [comment]
I've released a new version (1.16.0) with these changes: 
https://github.com/cesarParra/apexdocs/releases/tag/v1.16.0
https://www.npmjs.com/package/@cparra/apexdocs/v/1.16.0

**Pull request merged**